java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java

```diff
@@ -194,6 +180,50 @@ public void release(String lockId) {
         }
     }
 
+    private static void updateLabelFromSecret(SecretManagerServiceClient client, SecretName secretName, String label, String labelValue) {
+        try {
+            client.updateSecret(UpdateSecretRequest.newBuilder()
+                    .setSecret(Secret.newBuilder()
+                            .setName(secretName.toString())
+                            // Label format is https://cloud.google.com/compute/docs/labeling-resources#requirements
+                            .putLabels(label, labelValue)
+                            .build())
+                    .setUpdateMask(FieldMask.newBuilder()
+                            .addPaths("labels")
+                            .build())
+                    .build());
+        } catch (Exception e) {
+            log.log(Level.SEVERE, String.format("Cannot put the label of the version on the secret %s", secretName.toString()), e);
+            throw e;
+        }
+    }
+
+    private static void disableOldSecretVersions(SecretManagerServiceClient client, SecretName secretName, SecretVersion exceptVersion) {
+        try {
+            SecretManagerServiceClient.ListSecretVersionsPage enabledVersions = client.listSecretVersions(ListSecretVersionsRequest.newBuilder()
+                            .setFilter("state:ENABLED")
+                            .setParent(secretName.toString())
+                            // Reduce the page, as each version will be disabled one by one
+                            .setPageSize(NUMBER_OF_VERSIONS_TO_RETRIEVE)
+                            .build())
+                    .getPage();
+
+            // Disable secret version, just the first page
+            enabledVersions.getValues().forEach(i -> {
+                if (!i.getName().equals(exceptVersion.getName())) {
+                    client.disableSecretVersion(DisableSecretVersionRequest.newBuilder()
+                            .setName(i.getName())
+                            .build());
+
+                    log.info(String.format("Property: %s, disabled version %s", secretName, i.getName()));
+                }
+            });
+        } catch (Exception e) {
+            log.log(Level.SEVERE, String.format("Cannot disable old versions from secret %s", secretName.toString()), e);
+            throw e;
+        }
```

Comment:
again, disabling the secrets is a secondary action that in my option shouldn't jeopardize the main one `putConfigProperty`
access errors like not enough permissions should fail as it clearly states misconfiguration, but transient errors should be ignored. Next run will attempt to disable the secrets again.